modules/packages/Crypto.chpl

```diff
@@ -0,0 +1,876 @@
+                                keylen: c_int,
+                                outx: c_ptr(c_uchar)): c_int;
+  extern proc RAND_seed(const buf: c_void_ptr, num: c_int);
+
+
+  pragma "no doc"
+  proc generateKeys(bits: int) {
+   var localKeyPair: EVP_PKEY_PTR;
+   var keyCtx = EVP_PKEY_CTX_new_id(EVP_PKEY_RSA: c_int,
+                                    c_nil: ENGINE_PTR);
+   EVP_PKEY_keygen_init(keyCtx);
+   EVP_PKEY_CTX_set_rsa_keygen_bits(keyCtx, bits: c_int);
+   EVP_PKEY_keygen(keyCtx, localKeyPair);
+
+   EVP_PKEY_CTX_free(keyCtx);
+   return localKeyPair;
+  }
+
+  class CryptoBuffer {
+    var _len: int = 0;
+    var buffDomain: domain(1);
+    var buff: [buffDomain] uint(8);
+
+    /* The `CryptoBuffer` class constructor that initializes the buffer
+       when a `string` is supplied to it.
+
+       :arg s: `string` input for buffer conversion.
+       :type s: `string`
+
+       :return: An object of class `CryptoBuffer`.
+       :rtype: `CryptoBuffer`
+    */
+    proc CryptoBuffer(s: string) {
+      this._len = s.length;
+      this.buffDomain = {1..this._len};
+      for i in this.buffDomain do {
+        this.buff[i] = ascii(s[i]);
+      }
+    }
+
+    /* The `CryptoBuffer` class constructor that initializes the buffer
+       when a `[] uint(8)` is supplied to it.
+
+       :arg s: `[] uint(8)` input for buffer conversion.
+       :type s: `[] uint(8)]`
+
+       :return: An object of class `CryptoBuffer`.
+       :rtype: `CryptoBuffer`
+    */
+    proc CryptoBuffer(s: [] uint(8)) {
+      this._len = s.size;
+      this.buffDomain = s.domain;
+      for i in this.buffDomain do {
+        this.buff[i] = s[i];
+      }
+    }
+
+    /* Returns the entire internal buffer with each byte in ASCII.
+
+       :return: the internal buffer representation.
+       :rtype: `[] uint(8)`
+    */
+    proc getBuffData() {
+      return this.buff;
+    }
+
+    /* Returns the pointer to the entire internal buffer with
+       each byte in ASCII.
+
+       :return: pointer to the internal buffer representation.
+       :rtype: `c_ptr([] uint(8))`
+    */
+    proc getBuffPtr() {
+      return c_ptrTo(this.buff);
+    }
+
+    /* Returns the length of the entire internal buffer.
+
+       :return: length of the internal buffer represenation.
+       :rtype: `int`
+    */
+    proc getBuffSize(): int {
+      return this._len;
+    }
+
+    /* Returns the hexadecimal array representation of the entire internal
+        buffer.
+
+       :return: hex array representation of the internal buffer.
+       :rtype: `[] string`
+    */
+    proc toHex() {
+      var buffHex: [this.buffDomain] string;
+      for i in this.buffDomain do {
+        buffHex[i] = "%xu".format(this.buff[i]);
+      }
+      return buffHex;
+    }
+
+    /* Returns the hexadecimal string representation of the entire internal
+        buffer.
+
+       :return: hex string representation of the internal buffer.
+       :rtype: `string`
+    */
+    proc toHexString() {
```

Comment:
I think this should have a leading 0 for single digit hex values.  You can see the impact of it not being there in the buffer.chpl test below.  Line 6 in buffer.good should print as `0102030405`, not `12345`.